### ISSUE TITLE
syscheck/rootcheck during OS patching

### ISSUE THREAD

**Issue by aremai** [ossec/ossec-hids#677]
as I understand ossec's syscheck module regularly performs file integrity checks on files and compares the current hash with a known good hash for the file. if the hash doesn't match then ossec repots a change to the file....
the database of known good hashes is initally created the first time syscheck runs after ossec (agent/master) is installed. so far so good.

what happens after applying OS patches to the system? when you upgrade your system a lot of packages, e.g binaries will be updated and ergo their hashes will also change.
how to distinguish between a normal OS upgrade and a rootkit?

what is the recommended procedure during OS patching/upgrading?
how to avoid hundreds of alerts? would you just simply stop OSSEC?

**Comment by mstarks01** [comment]
Yes, you'll get a lot of alerts unless you tune them out. But this isn't a defect. Please try your question on the mailing list. Others have come up with creative solutions for this problem.

**Comment by aremai** [comment]
Hi Michael,

I've already tried the mailing list.... @ddpbsd said "There's been some thoughts on how to handle this, but no conclusions and no actions as of yet."

I thought there was more of a "general way" to avoid getting flooded with alerts during patching. Something like a best-practice or so.

I think the main issue during patching (correct me if I'm wrong) is the file integrity checks, because obviously the hashes change after patching. Log analysis wouldn't be such an issue I suppose.

**Comment by mstarks01** [comment]
IMO, the right way to do this would be to decode hashes from the syscheck alerts and add support to CDB such that the hashes could be stored in the CDB database. Then, a rule could be written to check file changes against that database and optionally not alert.

**Comment by aremai** [comment]
Sounds rather complicated. Is this something that could be integrated permanently into OSSEC? Surely this is something that a lot of users are facing. I don't think I'm the only one patching my machines ;)

**Comment by ddpbsd** [comment]
When I find time I'm hoping to add something to analysisd to allow a user
to check a cdb before creating a syscheck alert. Creating the db will be up
to the user though.
